### PR TITLE
spring cloud azure eventhubs samples: integration + binder + multibinder

### PR DESCRIPTION
**Pull request by jialigit**
azure spring integration sample eventhubs

### CODE REVIEW

File: eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/README.md [hunk 1]
```diff
@@ -44,13 +44,15 @@ Running this sample will be charged by Azure. You can check the usage and bill a
...
-            connection-string: [eventhub-namespace-connection-string]
-            checkpoint-storage-account: [checkpoint-storage-account]
-            checkpoint-access-key: [checkpoint-access-key]
-            checkpoint-container: [checkpoint-container]
+        cloud:
+          azure:
+              eventhubs:
+                  connection-string: [connection-string]
+                  processor:
+                      checkpoint-store:
```
Comment (yiliuTo) [inline eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/README.md:52]: identation is wrong

Comment (yiliuTo) [reply eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/README.md:52]: let's change the directory to new artifact id, and don't forget to modify the link path in readme.

File: eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/pom.xml [hunk 2]
```diff
@@ -2,59 +2,40 @@
...
+        <artifactId>azure-spring-boot-samples</artifactId>
+        <version>1.0.0</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
 
-  <artifactId>azure-spring-integration-sample-eventhubs</artifactId>
-  <groupId>com.azure.spring</groupId>
-  <version>1.0.0</version> <!-- {x-version-update;com.azure.spring:azure-spring-integration-sample-eventhubs;dependency} -->
-  <name>Azure Spring Integration Sample Event Hubs</name>
+    <artifactId>azure-spring-integration-sample-eventhubs</artifactId>
```
Comment (yiliuTo) [inline eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/pom.xml:12]: Change the artifact id and name here.

File: eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/pom.xml [hunk 3]
```diff
@@ -2,59 +2,40 @@
...
-    <groupId>org.springframework.boot</groupId>
-    <artifactId>spring-boot-starter-parent</artifactId>
-    <version>2.5.4</version> <!-- {x-version-update;org.springframework.boot:spring-boot-starter-parent;external_dependency} -->
-  </parent>
-  <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.azure.spring</groupId>
+        <artifactId>azure-spring-boot-samples</artifactId>
+        <version>1.0.0</version>
+    </parent>
```
Comment (yiliuTo) [inline eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/pom.xml:9]: add this: <relativePath>../../../pom.xml</relativePath>

File: eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/README.md (hunk 1 above)
Comment (yiliuTo) [reply eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/README.md:52]: Remove this sentense: "Please note `Basic` tier is unsupported." in this file.

**Review by yiliuTo — CHANGES_REQUESTED** [review]

File: eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/README.md (hunk 1 above)
Comment (jialigit) [reply eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/README.md]: updated

File: eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/pom.xml (hunk 3 above)
Comment (jialigit) [reply eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/pom.xml]: updated

File: eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/pom.xml (hunk 2 above)
Comment (jialigit) [reply eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-integration/pom.xml]: updated

File: eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md [hunk 4]
```diff
@@ -26,31 +26,27 @@ Running this sample will be charged by Azure. You can check the usage and bill a
...
 
 1.  Create [Azure Event Hubs][create-event-hubs].
-    Please note `Basic` tier is unsupported. After creating the Azure Event Hub, you
+    After creating the Azure Event Hubs, you
     can create your own Consumer Group or use the default "$Default" Consumer Group.
 
 1.  Create [Azure Storage][create-azure-storage] for checkpoint use.
 
-1.  **[Optional]** if you want to use service principal, please follow
-    [create service principal from Azure CLI][create-sp-using-azure-cli] to create one.
```
Comment (yiliuTo) [inline eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md:35]: We can retains this part

File: eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md [hunk 5]
```diff
@@ -26,31 +26,27 @@ Running this sample will be charged by Azure. You can check the usage and bill a
...
-
-1.  **[Optional]** if you want to use managed identity, please follow
-    [create managed identity][create-managed-identity] to set up managed identity.
-
```
Comment (yiliuTo) [inline eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md:39]: We can retain this part and udpate the application-sp/application-mi configuration files.

**Review by yiliuTo — CHANGES_REQUESTED** [review]

File: eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md (hunk 5 above)
Comment (jialigit) [reply eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md:39]: I'd update this after verified

File: README.md [hunk 6]
```diff
@@ -32,11 +32,9 @@
...
-| eventhubs        | [azure-spring-cloud-starter-eventhubs:2.9.0]                                                   | [eventhubs-operation](eventhubs/azure-spring-cloud-starter-eventhubs/eventhubs-operation)                                         |
-| eventhubs        | [azure-spring-cloud-stream-binder-eventhubs:2.9.0]                                             | [eventhubs-binder](eventhubs/azure-spring-cloud-stream-binder-eventhubs/eventhubs-binder)                                         |
-| eventhubs        | [azure-spring-cloud-stream-binder-eventhubs:2.9.0]                                             | [eventhubs-multibinders](eventhubs/azure-spring-cloud-stream-binder-eventhubs/eventhubs-multibinders)                             |
+| eventhubs        | [spring-cloud-azure-starter-integration-eventhubs:4.0.0-beta.1]                                                   | [eventhubs-integration](eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration)                                     |
```
Comment (yiliuTo) [inline README.md:35]: please update the link tags in the bottom of this file as well, and the link value should be udpated after release.

**Review by yiliuTo — CHANGES_REQUESTED** [review]

File: eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md (hunk 5 above)
Comment (jialigit) [reply eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md:39]: updated

File: README.md (hunk 6 above)
Comment (jialigit) [reply README.md]: updated

**Review by yiliuTo — CHANGES_REQUESTED** [review]

File: eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+   service principal or managed identity, update the `application-sp.yaml` or
+   `application-mi.yaml` respectively.
+    ```yaml
+    spring:
+        cloud:
+            azure:
+                eventhubs:
+                    connection-string: [connection-string]
+                    processor:
+                        checkpoint-store:
```
[the remark this answers is not in the document]
Comment (jialigit) [reply eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md:54]: updated

File: eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+1.  Update [application-mi.yaml][application-mi.yaml].
+    ```yaml
+    spring:
+      cloud:
+        azure:
+          credential:
+            managed-identity-client-id: [ managed-identity-client-id ]
+          profile:
+            tenant-id: [ tenant-id ]
+          #        msi-enabled: true
```
[the remark this answers is not in the document]
Comment (jialigit) [reply eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md:116]: updated

File: eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/pom.xml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+            <artifactId>spring-boot-starter-logging</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.hibernate.validator</groupId>
+            <artifactId>hibernate-validator</artifactId>
+        </dependency>
+<!--        <dependency>-->
+<!--            <groupId>org.springframework.boot</groupId>-->
+<!--            <artifactId>spring-boot-properties-migrator</artifactId>-->
+<!--            <scope>runtime</scope>-->
```
[the remark this answers is not in the document]
Comment (jialigit) [reply eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/pom.xml:38]: updated

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/resources/application-batch.yaml [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+            container-name: [container-name]
+            account-name: [account-name]
+            account-key: [account-key]
+    stream:
+      function:
+        definition: consume;supply
+      bindings:
+
+        consume-in-0:
+          content-type: text/plain # string message must specify this value
```
[the remark this answers is not in the document]
Comment (jialigit) [reply eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/resources/application-batch.yaml:17]: updated

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/resources/application-batch.yaml [hunk 11]
```diff
@@ -0,0 +1,38 @@
+spring:
+  cloud:
+    azure:
+      eventhubs:
+        connection-string:  [connection-string]
```
[the remark this answers is not in the document]
Comment (jialigit) [reply eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/resources/application-batch.yaml:5]: updated

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/BatchProducerAndConsumerConfiguration.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+                LOGGER.info("New message received: '{}', partition key: {}, sequence number: {}, offset: {}, enqueued time: {}",
+                        message.getPayload().get(i),
+                        ((List<Object>) message.getHeaders().get(EventHubsHeaders.PARTITION_KEY)).get(i),
+                        ((List<Object>) message.getHeaders().get(EventHubsHeaders.SEQUENCE_NUMBER)).get(i),
+                        ((List<Object>) message.getHeaders().get(EventHubsHeaders.OFFSET)).get(i),
+                        ((List<Object>) message.getHeaders().get(EventHubsHeaders.ENQUEUED_TIME)).get(i));
+            }
+
+
+       /*     Checkpointer checkpointer = (Checkpointer) message.getHeaders().get(CHECKPOINTER);
```
[the remark this answers is not in the document]
Comment (jialigit) [reply eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/BatchProducerAndConsumerConfiguration.java:51]: updated

File: eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+We have several ways to config the Spring Integration for Service
+Bus. You can choose anyone of them.
+
+>[!Important]
+>
+>  When using the Restful API to send messages, the **Active profiles** must contain `manual`.
+> 
+#### Method 1: Connection string based usage
+
+1. Update [application.yaml]. If you choose to use
```
[the remark this answers is not in the document]
Comment (jialigit) [reply eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md:44]: updated

File: eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+1.  Update event hub name and consumer group in
+    [ReceiveController][receive-controller] and [SendController][send-controller].
+
+#### Method 2: Service principal based usage
+
+1. Create a service principal for use in by your app. Please follow
+    [create service principal from Azure CLI][create-sp-using-azure-cli].
+
+2. Add Role Assignment for Event Hubs. See
+    [Service principal for Azure resources with Event Hubs][role-assignment]
```
Comment (yiliuTo) [inline eventhubs/spring-cloud-azure-starter-integration-eventhubs/eventhubs-integration/README.md:67]: no link here, please check carefully before you get your pr reviewed.

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/README.md [hunk 15]
```diff
@@ -5,226 +5,123 @@ languages:
...
 One way is to expose a Restful API to receive string message, another way is to automatically provide string messages.
-These messages are published to an event hub. The sample will also consume messages from the same event hub.
+These messages are published to one `Event Hub` instance and then consumed by one consumer 
+endpoint from the same application.
 
 ## Getting started
 
 Running this sample will be charged by Azure. You can check the usage and bill at 
-[this link][azure-account].
+[here][azure-account].
```
Comment (yiliuTo) [inline eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/README.md:24]: please remove "at".

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/pom.xml [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+            <artifactId>spring-boot-starter-actuator</artifactId>
+        </dependency>
+        <!-- TEST-->
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+    <build>
```
Comment (yiliuTo) [inline eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/pom.xml:50]: The `<build>` part can be deleted, same for other poms

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/resources/application.yaml [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+          consumer:
+            batch-mode: false # false is default value
+        supply-out-0:
+          destination: ${SAME_AS_ABOVE_DESTINATION}
+      eventhubs:
+        bindings:
+          consume-in-0:
+            consumer:
+              checkpoint:
+                mode: MANUAL # RECORD | MANUAL
```
Comment (yiliuTo) [inline eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-binder/src/main/resources/application.yaml:27]: Shouldn't Eventhubs support 4 types of checkpoint modes?

File: eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-multibinders/pom.xml [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+        </dependency>
+        <!-- TEST-->
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
```
Comment (yiliuTo) [inline eventhubs/spring-cloud-azure-stream-binder-eventhubs/eventhubs-multibinders/pom.xml:45]: same here

File: pom.xml [hunk 19]
```diff
@@ -20,7 +20,7 @@
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
     <version.spring.cloud>2020.0.3</version.spring.cloud>
-    <version.spring.cloud.azure>4.0.0-beta.1</version.spring.cloud.azure>
+    <version.spring.cloud.azure>4.0.0-beta.2</version.spring.cloud.azure>
```
Comment (yiliuTo) [inline pom.xml:23]: why do you change this?

File: keyvault/azure-securtiy-keyvault-jca/run-with-command-line-client-side/README.md [hunk 20]
```diff
@@ -49,7 +49,7 @@ jdk 11.0.12 or above
    ```
 1. Get the azure-security-keyvault-jca.jar. You can download the latest published jar from maven
    repository [azure-security-keyvault-jca][azure-security-keyvault-jca]. When this document is 
-   written, the latest jar is azure-security-keyvault-jca.2.1.0.jar
+   written, the latest jar is azure-security-keyvault-jca.2.2.0.jar
```
Comment (yiliuTo) [inline keyvault/azure-securtiy-keyvault-jca/run-with-command-line-client-side/README.md:52]: why do you change this?

File: pom.xml [hunk 21]
```diff
@@ -42,47 +42,48 @@
...
-    <module>aad/spring-cloud-azure-starter-active-directory/aad-web-application-and-resource-server</module>
-    <module>aad/spring-cloud-azure-starter-active-directory-b2c/aad-b2c-web-application</module>
-    <module>aad/spring-cloud-azure-starter-active-directory-b2c/aad-b2c-resource-server</module>
+    <module>aad/azure-spring-boot-starter-active-directory/aad-resource-server</module>
+    <module>aad/azure-spring-boot-starter-active-directory/aad-resource-server-by-filter</module>
+    <module>aad/azure-spring-boot-starter-active-directory/aad-resource-server-by-filter-stateless</module>
+    <module>aad/azure-spring-boot-starter-active-directory/aad-resource-server-obo</module>
+    <module>aad/azure-spring-boot-starter-active-directory/aad-web-application</module>
+    <module>aad/azure-spring-boot-starter-active-directory/aad-web-application-and-resource-server</module>
+    <module>aad/azure-spring-boot-starter-active-directory-b2c/aad-b2c-web-application</module>
```
Comment (yiliuTo) [inline pom.xml:51]: same here

**Review by yiliuTo — CHANGES_REQUESTED** [review]

**Comment by jialigit** [comment]
replaced with another pr

**Pull request closed without merging**